packages/usa-file-input/src/index.js

```diff
@@ -224,34 +287,89 @@ const removeOldPreviews = (dropTarget, instructions, inputAriaLabel) => {
   // Get rid of existing previews if they exist, show instructions
   if (filePreviews !== null) {
     if (instructions) {
-      instructions.classList.remove(HIDDEN_CLASS);
+      instructions.removeAttribute("hidden");
     }
-    fileInputElement.setAttribute("aria-label", inputAriaLabel);
     Array.prototype.forEach.call(filePreviews, removeImages);
   }
 };
 
+/**
+ * Update the screen reader-only status message after interaction
+ *
+ * @param {HTMLDivElement} statusElement - The screen reader-only container for file status updates.
+ * @param {Object} fileNames - The selected files found in the fileList object.
+ * @param {Array} fileStore - The array of uploaded file names created from the fileNames object.
+ */
+const updateStatusMessage = (statusElement, fileNames, fileStore) => {
+  const statusEl = statusElement;
+  let statusMessage = DEFAULT_FILE_STATUS_TEXT;
+
+  // If files added, update the status message with file name(s)
+  if (fileNames.length === 1) {
+    statusMessage = `You have selected the file: ${fileStore}`;
+  } else if (fileNames.length > 1) {
+    statusMessage = `You have selected ${
+      fileNames.length
+    } files: ${fileStore.join(", ")}`;
+  }
+
+  // Add delay to encourage screen reader readout
+  setTimeout(() => {
+    statusEl.textContent = statusMessage;
+  }, 1000);
+};
+
+/**
+ * Show the preview heading, hide the initial instructions and
+ * Update the aria-label with new instructions text
+ *
+ * @param {HTMLInputElement} fileInputEl - The input element.
+ * @param {Object} fileNames - The selected files found in the fileList object.
+ */
+const addPreviewHeading = (fileInputEl, fileNames) => {
```

Comment:
This function adds text describing:
- How many files
- Which files have been uploaded

So far I've only gotten it to read after initial upload. If I move to another area and comeback it doesn't get read on VoiceOver Chromium 113.

### Possible Future enhancements:
1. Add a unique ID to this message and tie it to the component via `aria-labelledby` [[MDN on aria-labelledby]](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby)
2. Move the preview heading text so it always gets read.
3. Dynamically update the aria-label
    
**Move the preview heading text example**

https://github.com/uswds/uswds/assets/3385219/e9cebeef-9ba5-4eab-9f68-5c83293f61e6

Reply:
Thanks @mejiaj. I've opened https://github.com/uswds/uswds/issues/5304 to start looking into this. I will begin investigating as a fast follow-up.